File: 프로그래머스/3/42884. 단속카메라/단속카메라.py
def solution(routes):
    answer = 0
    routes = sorted(routes, key=lambda x:(x[0],x[1]))
    c = -30050
    for s, e in routes:
        if c < s:
            c = e
            answer +=1
        else:
            if e < c:
                c = e
    return answer

# print(solution([[-20,15], [-14,-5], [-18,-13], [-5,-3]])) #2

File: 프로그래머스/3/42884. 단속카메라/단속카메라.js
function solution(routes) {
    var answer = 0;
    routes = routes.sort((a,b)=>{
        if(a[0]==b[0]) return a[1]-b[1]
        return a[0]-b[0]
    })
    
    let s = -30002
    let e = -30001
    
    for(let [start, end] of routes){
        if(e < start){
            answer++
            s = start
            e = end
        }else{
            s = start
            e = Math.min(end,e)
        }
    }
    return answer;
}